frontend/device/remarkable/device.lua

```diff
@@ -125,5 +126,159 @@ function Remarkable:reboot()
     os.execute("systemctl reboot")
 end
 
+function os.capture(cmd, raw)
+    local f = io.popen(cmd, 'r')
+    if not f then
+        return nil
+    end
+    local s = f:read('*a')
+    f:close()
+    if not s then
+        return nil
+    end
+    if raw then
+        return s
+    end
+    s = string.gsub(s, '^%s+', '')
+    s = string.gsub(s, '%s+$', '')
+    s = string.gsub(s, '[\n\r]+', ' ')
```

Comment:
I'm not sure if I'd expect those things to be filtered out by default. Could you clarify the thought process?

Reply:
https://stackoverflow.com/a/326715
Basically it's removing extra whitespace and newlines. I left it in since I don't expect extra newlines in my results anyways.

Reply:
Involving a particular set of results in the thought process seems contradictory to putting it on `os.*`.

Taking a peek around the code it looks like we normally just use something simple like this, but I can see how that'd be annoyingly repetitive for your use case here.

```lua
local std_out = io.popen("rot wifi")
if std_out then
    local result = std_out:read("*all")
    std_out:close()
end
```

This function might be useful in util, but it seems to me that the unfiltered output is what you'd expect by default. What do you guys think @poire-z @NiLuJe @pazos?

(If you just keep it as a local here the logic should be fine, but it feels like it could have some generic use.)

Reply:
> but it seems to me that the unfiltered output is what you'd expect by default. 

To me too. And it looks like with how it is used here, it should be fine with leading spaces or `\n`: tonumber() is fine with them, I guess it's still valid json so rapidjson should be fine too. May be just have isempty() checks also for `"\n"`.

> it feels like it could have some generic use

Well, we might want to strip some but not all, so, that would need many parameters :) I prefer the strippings done locally, so it can be more easily customized (and we won't go adding one parameter when we need to strip something else :)

Reply:
So something like this? Or keep it completely local here (but still in roughly this form).

```lua
function util.osCapture(cmd)
    local f = io.popen(cmd, "r")
    if not f then
        return nil
    end
    local s = f:read("*a")
    f:close()
    if not s then
        return nil
    end
    return s
end
```

Reply:
util.stdout?

Reply:
Or `util.somethingPopen`/`util.popenSomething`.

Reply:
`util.getCommandOutput(cmd, stripped)`
(with may be `stripped=true/false` just to remove leading and trailing spaces)

The first 3 lines are not needed in there, are they ?
```
    if not s then
        return nil
    end
    return s
```

Reply:
Might as well have util.getFileContent(path) as we have a few things like:
https://github.com/koreader/koreader/blob/27fe0d7957fd48c375cdae48bc3bf30ceef1edeb/frontend/apps/reader/modules/readerdictionary.lua#L71-L80

Reply:
I don't know, depends on if io.popen could return `false` and if we would want to hide that. But indeed it looks unnecessary.